compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -266,7 +266,11 @@ private void defineConstructs(BLangPackage pkgNode, SymbolEnv pkgEnv) {
             for (BLangImportPackage unresolvedPkg : importHolder.unresolved) {
                 BPackageSymbol pkgSymbol = importHolder.resolved.symbol; // get a copy of the package symbol, add
                                                                          // compilation unit info to it,
-    
+                if (pkgSymbol == null) {
```

Comment:
Can't this be checked in the outer for loop? Or we are repeating logic here for each unresolvedPkg?

Reply:
Moved `BPackageSymbol pkgSymbol = importHolder.resolved.symbol;` to the outer loop.  
We need that check inside the inner loop to printout the message.